### PR TITLE
Vu sprint1 review

### PR DESCRIPTION
**Pull request by vumd-UC**
Sprint Review for Sprint 1

### CODE REVIEW

File: BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt [hunk 1]
```diff
@@ -5,10 +5,15 @@ import edu.uc.brown5a7.BottomsUp.service.DrinkService
...
 import org.junit.Assert
+import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.junit.rules.TestRule
 
+/*
+ * Integration test for Drinkservice.kt. Tests fetching of drinks based on search terms and search categories.
+ */
+
```
Comment (vumd-UC) [inline BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt:16]: Added in Javadoc for integration test.

File: BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt [hunk 2]
```diff
@@ -17,19 +22,26 @@ class DrinkDataIntegrationTest {
 
     var drinkService = mockk<DrinkService>()
 
+    @Before
+    fun setUp()
+    {
+        mvm = MainViewModel()
+    }
+
```
Comment (vumd-UC) [inline BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt:30]: Used the @Before to create a new MainViewModel before every test. Before, creating a mvm was being called as a given for every test. Removes a bit of redundancy.

File: BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt [hunk 3]
```diff
@@ -17,19 +22,26 @@ class DrinkDataIntegrationTest {
...
+    {
+        mvm.fetchDrinks(searchTerm)
     }
 
-    private fun whenSearchForCoke() {
-        mvm.fetchDrinks("coke")
+    private fun whenSearchingForCatagory(searchCatagory: String)
+    {
+        mvm.fetchCategory(searchCatagory)
     }
```
Comment (vumd-UC) [inline BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt:45]: Added in functions that use the fetchDrinks and fetchCategory functions that pass in a parameter. Removes the need for multiple when functions to search for a specific term

**Comment by vumd-UC** [comment]
- Analysis: Solid framework for the eventual end goal. Files and functions all make sense, as well as overall program flow.
- Was the program available in UC Github on time?
--Yes, had a bit of issues with permissions to write, but got resolved quickly.
- Is the program documented/commented well enough for you to understand?
--Yes, code is easy to understand.
- Does the program compile?
--Yes, but not in an intuitive way. @robertdhart99 @brownadam010 @mooncakes-dev You will most definitely want to take a look at refactoring your code to have it not run through the nested folders of BottomsUp/BottomsUp. Android studio will fail to pick up anything to build if running through the top level BottomsUp, but will run through if running through the bottom level BottomsUp. This isn't really intuitive for anyone looking in, and people might give false alarms on the application not running.
- Rationale behind your changes.
--In comments within the file changed. TL;DR : Added in class level javadoc for integration test, removed redundant given statements by implementing an @Before, changed when functions to search using a given parameter rather than hard coding multiple functions.

Three specific technical concepts from this review:

1.  Extending/Overriding RecyclerView
drinkAdapter overrides several methods from the RecyclerView, and gives me a better idea of how to override within Kotlin.
2. Casting variables to certain types in Kotlin
`drinks = drinkService.fetchDrinks("blank")
        val drinksFullList: ArrayList<Drink> = drinks as ArrayList<Drink>`
Didn't know how to cast variables within Kotlin
3. Using MutableLiveData
While I had an understanding of it from class, going through the code for BottomsUp gave me a better understanding of how it works.

**Review by vumd-UC — COMMENTED** [review]
Added in comments at file level for what changed.

**Comment by robertdhart99** [comment]
Thank you for the feedback. It was very informative and helpful.

File: BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt (hunk 2 above)
Comment (discospiff) [reply BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt:30]: Hmmm, good idea.

File: BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt (hunk 3 above)
Comment (discospiff) [reply BottomsUp/app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt:45]: "Category" is the proper spelling.

**Comment by discospiff** [comment]
Some good suggestions here... it looks like the team has already merged them.

**Pull request merged**